tox.ini

```diff
@@ -2,20 +2,82 @@
 min_version = 4.0
 env_list = py38,py39,py310,py311,py312
 
+
 [testenv]
-deps = .[dev]
+description = Run all tests
+deps =
+    .[dev]
+commands =
+    pytest tests/ {posargs}
+
+
+[testenv:test-unit]
+description = Run unit tests
+deps =
+    .[dev]
+commands =
+    python -m pytest tests/unit {posargs}
+
+
+[testenv:test-integration]
+description = Run integration tests
+deps =
+    .[dev]
 commands =
-    python -m pytest tests/unit
+    python -m pytest tests/integration {posargs}
+
 
-[testenv:lint]
+[testenv:test-e2e]
+description = Run end-to-end tests
+deps =
+    .[dev]
 commands =
-    python -m ruff check src tests
+    python -m pytest tests/e2e {posargs}
 
-[testenv:formatting]
+
+[testenv:quality]
+description = Run all quality checks
+deps =
+    .[dev]
 commands =
-    python -m black --check src tests
-    python -m isort --check src tests
+    ruff check src tests
+
+
+[testenv:style]
+description = Run style checks and fixes
+deps =
+    .[dev]
+commands =
+    ruff format src tests
+
 
 [testenv:types]
+description = Run type checks
+deps =
+    .[dev]
+commands =
+    mypy --check-untyped-defs
+
+
+[testenv:build]
+description = Build the project
+deps =
+    setuptools
+    wheel
+commands =
+    python setup.py sdist bdist_wheel
```

Comment:
python also supports `python -m build`. Does it make the same?

Reply:
Good question, I believe the only difference is the setup.py pathway doesn't isolate the build environment. python -m is the newer recommended way, so I'll update to that